geopyspark-backend/geotrellis/src/main/scala/geopyspark/geotrellis/protobufs/TupleProtoBuf.scala

```diff
@@ -50,6 +74,17 @@ trait TupleProtoBuf {
         multibandTileProtoBufCodec.decode(message.tiles.get))
   }
 
+  implicit def tupleSpatialKeyImageProtoBufCodec =
+    new ProtoBufCodec[(SpatialKey, Array[Byte]), ProtoTuple] {
```

Comment:
Right the problem is the `projectedExtent = Some(keyCodec.encode(tuple._1))` where we expect to switch which field we set based on type. There is a way to work around this with lenses but yeah, that doesn't make sense for this PR.